content/docs/forwarding-refs.md

```diff
@@ -1,76 +1,76 @@
+**ref のフォワーディングはオプトインの機能で、それにより、コンポーネントは `ref` を取ることができるようになり、コンポーネントは `ref` をさらに下層の子に渡し（つまり、ref を "フォワーディング" し）ます。**
 
-In the example below, `FancyButton` uses `React.forwardRef` to obtain the `ref` passed to it, and then forward it to the DOM `button` that it renders:
+下の例では、`FancyButton` は渡された `ref` を取得し、それから描画する `button` DOM にフォワーディングするために、`React.forwardRef` を使っています。
 
 `embed:forwarding-refs/fancy-button-simple-ref.js`
 
-This way, components using `FancyButton` can get a ref to the underlying `button` DOM node and access it if necessary—just like if they used a DOM `button` directly.
+このように、`FancyButton` を使ったコンポーネントは下層の `button` DOM ノードの ref を取得することができ、必要であれば `button` DOM を直接使うかのように、DOM にアクセスすることができます。
 
-Here is a step-by-step explanation of what happens in the above example:
+上の例で、何が起こっているかを順々に説明します。
 
-1. We create a [React ref](/docs/refs-and-the-dom.html) by calling `React.createRef` and assign it to a `ref` variable.
-1. We pass our `ref` down to `<FancyButton ref={ref}>` by specifying it as a JSX attribute.
-1. React passes the `ref` to the `(props, ref) => ...` function inside `forwardRef` as a second argument.
-1. We forward this `ref` argument down to `<button ref={ref}>` by specifying it as a JSX attribute.
-1. When the ref is attached, `ref.current` will point to the `<button>` DOM node.
+1. `React.createRef` を呼び、[React ref](/docs/refs-and-the-dom.html) をつくり、それを `ref` 変数に代入します。
+1. `ref` を JSX の属性として指定することで `<FancyButton ref={ref}>` に渡します。
+1. React は 2 番目の引数として、`forwardRef` 内の関数に `ref` を渡します。
+1. この `ref` を JSX の属性として指定することで `<button ref={ref}>` に渡します。
+1. この ref が付与されると、`ref.current` は `<button>` DOM ノードのことを指すようになります。
 
->Note
+>補足
 >
->The second `ref` argument only exists when you define a component with `React.forwardRef` call. Regular function or class components don't receive the `ref` argument, and ref is not available in props either.
+> 2 番目の引数 `ref` は `React.forwardRef` の呼び出しを使ってコンポーネントを定義したときにだけ存在します。通常の関数またはクラスコンポーネントは `ref` 引数を受け取らず、ref は props 中でも使用することができません。
 >
->Ref forwarding is not limited to DOM components. You can forward refs to class component instances, too.
+> ref のフォワーディング先は DOM コンポーネントだけにとどまりません。クラスコンポーネントインスタンスへの ref もフォワーディングすることができます。
 
-## Note for component library maintainers {#note-for-component-library-maintainers}
+## コンポーネントライブラリのメンテナ向けの補足 {#note-for-component-library-maintainers}
 
-**When you start using `forwardRef` in a component library, you should treat it as a breaking change and release a new major version of your library.** This is because your library likely has an observably different behavior (such as what refs get assigned to, and what types are exported), and this can break apps and other libraries that depend on the old behavior.
+**コンポーネントライブラリの中で、`forawrdRef` を使い始めた場合、破壊的変更として扱い、ライブラリのメジャーバージョンをリリースすべきです。**ライブラリが今までと著しく違う挙動（例えば、何の ref が代入されるのか、どのような type がエクスポートされるかのような動作）をする可能性があるからです。
```

Comment:
言い回しの調整 [IMO] と `this can break apps and other libraries that depend on the old behavior.` の訳がなかったので足しました。

```suggestion
**コンポーネントライブラリの中で、`forawrdRef` を使い始めた場合、破壊的変更として扱い、新しいメジャーバージョンをリリースすべきです。**ライブラリが今までと著しく違う挙動（例えば、どの値が ref に代入されるかや、どの型がエクスポートされるのか）をする可能性があり、古い挙動に依存しているアプリケーションや他のライブラリを壊す可能性があるからです。
```

Reply:
失礼しました。カッコ内も悩んだところでした